types/function_list_builder.go

```diff
@@ -50,17 +94,38 @@ func (s *FunctionLookupBuilder) Build() (map[string][]string, error) {
 	marshalErr := json.Unmarshal(bytesOut, &functions)
 
 	if marshalErr != nil {
-		return map[string][]string{}, errors.Wrap(marshalErr, fmt.Sprintf("unable to unmarshal value: %q", string(bytesOut)))
+		return []types.FunctionStatus{}, errors.Wrap(marshalErr, fmt.Sprintf("unable to unmarshal value: %q", string(bytesOut)))
 	}
 
-	serviceMap := buildServiceMap(&functions, s.TopicDelimiter)
-
-	return serviceMap, err
+	return functions, nil
 }
 
-func buildServiceMap(functions *[]types.FunctionStatus, topicDelimiter string) map[string][]string {
+// Build compiles a map of topic names and functions that have
+// advertised to receive messages on said topic
+func (s *FunctionLookupBuilder) Build() (map[string][]string, error) {
```

Comment:
It would be tricky to add unit test here but it might be a good idea.

Reply:
Yes, Unit test was already there, I have updated it.